tests/backends/data/test_async_es.py

```diff
@@ -475,6 +475,37 @@ async def test_backends_data_async_es_read_with_query(es, async_es_backend, capl
     await backend.close()
 
 
+@pytest.mark.anyio
+async def test_backends_data_async_es_write_with_simultaneous(
+    es, async_es_backend, caplog
+):
+    """Test the `AsyncESDataBackend.write` method, given `simultaneous` set to `True`,
+    should insert the target documents concurrently.
+    """
+    backend = async_es_backend()
+    data = ({"value": str(idx)} for idx in range(5))
+
+    with caplog.at_level(logging.INFO):
+        assert (
+            await backend.write(
+                data, chunk_size=3, simultaneous=True, max_num_simultaneous=2
+            )
+            == 5
+        )
```

Comment:
Unless I am mistaken, this does not test the actual simultaneity. 

For `AsyncLRSHTTP`, there are timing-based tests that are flagged to be skipped but are intended to be kept for dev purpose (There is perhaps a better solution):

https://github.com/openfun/ralph/blob/0bfcd837f3501699ffd5d994af3656a8506a9bc9/tests/backends/http/test_async_lrs.py#L1009

Reply:
You are right) Maybe we should limit this test to only check that the `similtaneous` parameter is passed to the parent `write` method? The main testing of simultaneity is done in `test_backends_data_base_async_write_with_simultaneous`, however, indeed, for now, we do not test the timing (or order) of the write calls - would counting the number of times `gather_with_limited_concurrency` is called suffice (to avoid timing-based tests)? WDYT?

Reply:
Updated the `test_backends_data_base_async_write_with_concurrency` with a call counter for `gather_with_limited_concurrency`)